### ISSUE TITLE
install repo-supported version of docker

### ISSUE THREAD

**Issue by jodh-intel** [clearcontainers/tests#666]
The `.ci/setup_env_ubuntu.sh` and `.ci/setup_env_fedora.sh` scripts  install the `docker-ce` package if docker is not already installed. However, they should be installing the newest version of docker that Clear Containers supports for the repo they are testing.

For example, for the runtime this should be `v17.09-ce`. See `docker_version=` and `docker_swarm_version=` in:

- https://github.com/clearcontainers/runtime/blob/master/versions.txt

This would translate into something like the following (ignore actual version numbers below - it's just showing the package-manager command to install a named version):

# Ubuntu

```
$ sudo apt-get -y install docker-ce=17.05.0~ce-0
```

# Fedora

```
$ sudo dnf -y install docker-ce-17.09.0.ce-1.fc26
```

A complication is that the version of docker used for swarm may be different to this version. Today, swarm will only work with docker `1.12.1`.

This also might have a CI impact.

/cc @sboeuf, @mcastelino, @grahamwhaley.

**Comment by jodh-intel** [comment]
Darn. The number of the beast :japanese_ogre:. I should have raised this yesterday! :smile:

**Comment by jodh-intel** [comment]
Idea: we could create an `manage_ctr_mgr.sh` script:

```
$ manage_ctr_mgr.sh [docker|kubernetes] [install|list|configure|remove]
```

... which when run as:

```
$ manage_ctr_mgr.sh docker install
```

... would do the following:

- if docker already installed, `exit 0`.
- if docker not installed, attempt to find `versions.txt` and install `docker_version=`.

But if run as, say,

```
$ manage_ctr_mg.sh docker install --tag swarm --force
```

... it would:

- remove an existing docker (we'd need a regex to find `docker.io`, `docker-engine`, `docker-ce` packages, etc).
- attempt to find `versions.txt` and install `docker_${tag}_version=`.

It will need to deal with a few things:

- `sudo`.
- the `docker` group (maybe?)
- it will need a `remove` option.
- it could also have a `configure --runtime=...` option.
- it should call `logger -s -t $program_name" ...` for any operation that modifies the version of docker on the system (`install`, `configure`, `remove`) so we have a trace of what it did.
- also handle k8s! ;)

Thoughts?

**Comment by sboeuf** [comment]
@jodh-intel This proposal sounds good to me. All cases seem to be covered here!

**Comment by jodh-intel** [comment]
Fixed by #800.
